ceno_zkvm/src/bin/e2e.rs

```diff
@@ -148,7 +148,11 @@ fn main() {
     // do statistics
     let stat_recorder = StatisticRecorder::default();
     let transcript = TranscriptWithStat::new(&stat_recorder, b"riscv");
-    verifier.verify_proof(zkvm_proof.clone(), transcript).ok();
+    assert!(
+        verifier
+            .verify_proof_halt(zkvm_proof.clone(), transcript, zkvm_proof.has_halt())
```

Comment:
If you set `expect_halt = zkvm_proof.has_halt()` then the new check that you added in this pr (i.e. `if has_halt != expect_halt`) will always pass, right?

Reply:
per offline discussion, `verify_proof_halt ` is internal api, thus here halt is optional.
while in `verify_proof` it will always set `true` and assert halt happenend